keel-core/src/main/kotlin/com/netflix/spinnaker/keel/events/NotificationEvent.kt

```diff
@@ -1,13 +1,40 @@
 package com.netflix.spinnaker.keel.events
 
+import com.netflix.spinnaker.keel.api.DeliveryConfig
+import com.netflix.spinnaker.keel.core.api.EnvironmentArtifactPin
+import com.netflix.spinnaker.keel.core.api.PinnedEnvironment
 import com.netflix.spinnaker.keel.notifications.Notification
 import com.netflix.spinnaker.keel.notifications.NotificationScope
 import com.netflix.spinnaker.keel.notifications.NotificationType
 
-data class NotificationEvent(
-  val scope: NotificationScope,
-  val ref: String,
-  val type: NotificationType,
-  val message: Notification
-)
+abstract class NotificationEvent{
+  abstract val scope: NotificationScope
+  open val ref: String? = null
+  abstract val type: NotificationType
+  open val message: Notification? = null
+}
 
+data class UnhealthyNotification(
+  override val scope: NotificationScope,
+  override val ref: String?,
+  override val type: NotificationType,
+  override val message: Notification?
+  ) :NotificationEvent()
+
+data class PinnedNotification(
+  val config: DeliveryConfig,
+  val pin: EnvironmentArtifactPin
+): NotificationEvent() {
```

Comment:
```suggestion
) : NotificationEvent() {
```

Reply:
there is also some other wonky spacing in this file